gerritbot/examples/gerritbot-console.rs

```diff
@@ -5,46 +5,131 @@
 use std::io::{BufRead as _, BufReader, Write as _};
 use std::path::PathBuf;
 
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use serde_json;
 use structopt::StructOpt;
 
 use futures::{future, stream, sync::mpsc::channel, Future as _, Sink as _, Stream as _};
-use log::{error, info};
+use log::{error, info, warn};
 
 use gerritbot as bot;
 use gerritbot_gerrit as gerrit;
 use gerritbot_spark as spark;
 
 #[derive(StructOpt, Debug)]
+#[structopt(name = "gerritbot-console", rename_all = "kebab-case")]
+/// Run the gerritbot without actually connecting to Spark. Instead the
+/// program's stdin can be used to simulate sending messages. Replies will be
+/// sent to stdout. Log messages will only appear on stderr. If the --email
+/// option (see below) is given each input line will be treated as message from
+/// that user. Otherwise each line has to be prefixed with the email address of
+/// the intended sender, a colon (:) and an optional space character.
 struct Args {
-    /// Gerrit username
-    #[structopt(short = "u")]
+    /// Gerrit bot username
+    ///
+    /// This is the username the bot uses to connect to the Gerrit server.
+    #[structopt(short, long)]
     username: String,
     /// Gerrit hostname
+    ///
+    /// Address of the Gerrit server.
     hostname: String,
     /// Gerrit SSH port
-    #[structopt(short = "p", default_value = "29418")]
+    #[structopt(short, long, default_value = "29418")]
     port: u32,
     /// Path to SSH private key
-    #[structopt(short = "i", parse(from_os_str))]
-    private_key_path: PathBuf,
+    #[structopt(short, long, parse(from_os_str))]
+    identity_file: PathBuf,
     /// Enable verbose output
-    #[structopt(short = "v")]
+    #[structopt(short, long)]
```

Comment:
`conflicts_with` = "quiet"

Reply:
It's reflexive so the one below is enough.